bin/install

```diff
@@ -18,23 +19,109 @@ error_exit() {
   exit "${2:-1}"
 }
 
-platform() {
+get_platform() {
   local -r os=$(
     case "$(uname | tr '[:upper:]' '[:lower:]')" in
       darwin) echo "osx" ;;
       *) echo "linux" ;;
     esac
   )
 
-  echo "${os}-amd64"
+  echo "${os}"
+  return
 }
+get_arch(){
+  declare arch="$(uname -m)"
+  if [ "$arch" == 'x86_64' ]; then
+    echo '64'
+  elif [ "$arch" == 'i386' ]; then
+    echo '32'
+  elif [ "$arch" == 'i686' ]; then
+    echo '32'
+  else
+    error_exit 'There are no official releases for your architecture'
+  fi
+  return
+}
+
+get_assets_url() {
+  declare install_version="$1"
+
+  declare releases_json=$(curl -s "$RELEASES_URL")
+
+  declare -a asset_urls=($(echo $releases_json | grep -oP "(?<=\"assets_url\": \")[^\"]{8,}(?=\",)"))
+  declare -a tag_names=( $(echo $releases_json | grep -oP "(?<=\"tag_name\": \")[^\"]{3,}(?=\",)"))
+
+  for ((index=0; index<${#tag_names[@]}; index++)); do
+    if [ "${tag_names[$index]}" = "$install_version" ]; then
+      echo "${asset_urls[$index]}"
+      return
+    fi
+  done
+  error_exit 'Given version did not match any releases. Try list-all to see available options'
+}
+find_all_asset_names() {
```

Comment:
`find_all_asset_names(install_version)` calls `get_assets_url($install_version)` to get the endpoint for the release of the given version, then extracts the values of all 'name' properties within the JSON, and returns them in an array.

Due to the way arrays work in bash scripting, unexpected behaviors might happen is a release asset contains a space  in its name.